tools/deleteorphanlogcommand.class.php

```diff
@@ -0,0 +1,188 @@
+            $output,
+            new ConfirmationQuestion(
+               '<comment>' . __('Do you want to launch operation ?') . ' [yes/No]</comment>',
+               false
+            )
+         );
+         if (!$run) {
+            $output->writeln(
+               '<comment>' . __('Operation aborted.') . '</comment>',
+               OutputInterface::VERBOSITY_VERBOSE
+            );
+            return 0;
+         }
+      }
+
+      $this->deleteOrphan($input, $output);
+
+      return 0;
+   }
+
+   private function deleteOrphan(InputInterface $input, OutputInterface $output) {
+
+      $globalCount = 0;
+      $listTable = $this->db->listTables();
+      $dry_run = $input->getOption('dry-run');
+
+      while ($table = $listTable->next()) {
+         $tablename = $table['TABLE_NAME'];
+
+         try {
+            $itemtype = getItemTypeForTable($tablename);
+            $output->writeln(
+               '<comment>' . sprintf(__('Searching for orphaned "%s"...'), $itemtype) . '</comment>',
+               OutputInterface::VERBOSITY_QUIET
+            );
+         } catch (\Throwable $th) {
+            $output->writeln(
+               sprintf(
+                  '<error>' . __('Skip tablename %s : ERROR -> %s') . '</error>',
+                  $tablename,
+                  $th->getMessage()
+               )
+            );
+            continue;
+         }
+
+         $result = $this->db->request(
+            [
+               'SELECT' => ['id', 'items_id'],
+               'FROM' => 'glpi_logs',
+               'WHERE' => [
+                  ['NOT' => ['items_id' => new QuerySubQuery(['SELECT' => 'id', 'FROM' => $tablename])]],
+                  'itemtype' => $itemtype,
+               ],
+            ]
+         );
+
+         $globalCount += $result->numrows();
+
+         if ($result->numrows() > 0) {
+            $msg = $dry_run ? __('Found %s orphaned "%s" record(s)...') : __('Deleting %s orphaned "%s" record(s)...');
+            $output->writeln('<info>' . sprintf($msg, $result->numrows(), $itemtype) . '</info>');
+
+            if (!$dry_run) {
+               $progress_bar = new ProgressBar($output);
+
+               $i = $total = 0;
+               foreach ($result as $row) {
+                  $ids[] = $row['id'];
+                  $i++;
+                  $total++;
+                  if ($i % 1000 == 0 || count($result) === $total) {
+                     $this->db->delete('glpi_logs', ['id' => $ids]);
+                     $progress_bar->advance($i);
+                     //reset
+                     $ids = [];
+                     $i=0;
+                  }
+               }
+
+               $progress_bar->finish();
+               $this->output->write(PHP_EOL);
+            } else {
+               if ($dry_run) {
+                  $output->writeln(
+                     '<comment>' . __('Effective deletion disabled (remove --dry-run arg)') . '</comment>'
+                  );
+               }
+
```

Comment:
Should not be repeated for eahc table. See my next comment for suggestion.

```suggestion
            } else {
               if ($dry_run) {
                  $output->writeln(
                     '<comment>' . __('Effective deletion disabled (remove --dry-run arg)') . '</comment>'
                  );
               }

```